src/replace_landsurface_with_FF_IC.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 ACCESS-NRI (https://www.access-nri.org.au/)
+# See the top-level COPYRIGHT.txt file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Created by: Chermelle Engel <Chermelle.Engel@anu.edu.au>
+
+import iris
+import mule
+import os
+import sys
+import numpy as np
+from glob import glob
+from pathlib import Path
+import xarray as xr, sys, argparse
+from datetime import datetime,timedelta
+
+class ReplaceOperator(mule.DataOperator):
+    """ Mule operator for replacing the data"""
+    def __init__(self):
+        pass
+    def new_field(self, sources):
+        print('new_field')
+        return sources[0]
+    def transform(self, sources, result):
+        print('transform')
+        return sources[1]
+
+def replace_in_ff_from_ff(f, sf, mf_out, replace):
+
+    replacement_data = sf.get_data()
+    mf_out.fields.append(replace([f, replacement_data]))
+
+def swap_land_ff(mask_fullpath, ic_file_fullpath, source_fullpath,ic_date):
+    """
+    Function to get the land/surface data from another fields file into the start dump.
+
+    Parameters
+    ----------
+    mask_fullpath : Path
+        Path to the mask defining the spatial extent
+    ic_file_fullpath : string
+        Path to file with the coarser resolution data to be replaced with ".tmp" appended at end
+    ic_date : string
+        The date-time required in "%Y%m%d%H%M" format
```

Comment:
`source_fullpath` will need an entry in this parameter list

Reply:
Yes, I will add that in thanks Tommy.